python_src/src/lamp_py/performance_manager/l0_gtfs_rt_events.py

```diff
@@ -85,58 +85,73 @@ def combine_events(
     )
     process_logger.log_start()
 
+    trip_stop_columns = get_unique_trip_stop_columns()
+
     # merge together the trip stop hashes and the timestamps
     events = pandas.merge(
-        vp_events[["trip_stop_hash", "vp_stop_timestamp", "vp_move_timestamp"]],
-        tu_events[["trip_stop_hash", "tu_stop_timestamp"]],
+        vp_events[
+            trip_stop_columns + ["vp_stop_timestamp", "vp_move_timestamp"]
+        ],
+        tu_events[
+            trip_stop_columns
+            + [
+                "tu_stop_timestamp",
+            ]
+        ],
         how="outer",
-        on="trip_stop_hash",
+        on=trip_stop_columns,
         validate="one_to_one",
     )
 
-    # concat all of the details that went into each trip stop hash, dropping
+    # concat all of the details that went into each trip event, dropping
     # duplicates. this is now a dataframe mapping hashes back to the things
     # that went into them.
     # TODO: review trip_id processing # pylint: disable=fixme
     #  add more intelligent trip_id processing, this approach will randomly select trip_id record to keep
```

Comment:
I don't think so.

The issue is that we are joining events on our `trip_stop_columns` to get unique trip-stop records. For any columns  that are not `trip_stop_columns` it is possible that records with the same `trip_stop_columns` values could have different values for any other `details_columns` we are transferring to the RDS: 
- stop_sequence
- trip_id
- vehicle_label
- vehicle_consist

This might not currently be an issue, but it is a limitation of the methodology we're using to create our trip-stop events. I'm up for documenting in a different way, but I do think it needs to be noted as a possible issue.

Reply:
added as DRAGON comment